snuba/api.py

```diff
@@ -183,12 +183,14 @@ def parse_request_body(request):
 
 def validate_request_content(body, schema, timer):
     try:
-        schemas.validate(body, schema)
+        request = schema.validate(body)
     except jsonschema.ValidationError as error:
```

Comment:
I would not explicitly catch jsonschema.ValidationError anymore. In the previous implementation the only method in the try/except block was a direct call to jsonschema validate, thus it was pertinent to expect a jsonschema.ValidationError. Now the type of validation performed by the RequestSchema object is opaque to this method. I would either catch any exception or define a custom validation exception.

Reply:
I think this makes sense in general.

It's a bit tricky because the control flow for exceptions changed a bit while you were out with getsentry/snuba#420. The error payload here _is_ dependent on this being a JSON schema, which I think is appropriate. I'll put some thought into figuring out how this should work. Here's where the error is introspected and relevant payload is generated now: https://github.com/getsentry/snuba/blob/14ad5d85e008eb442e15b23b976d4cd46841a416/snuba/api.py#L77-L83

Reply:
damn, well ok then. I would advise creating a dedicated exception that may wrap the jsonschema.ValidationError but that's not needed in this PR